### DISCUSSION TITLE
lotus depth  is available

### DISCUSSION THREAD

**Discussion by loawizard** [nagadomi/nunif#237] [General]
https://github.com/EnVision-Research/Lotus

can we support this depthmap.

**Comment by kaelsonofkrypto** [comment]
I agree can, we please get this depth map. It looks like it may possibly be better than any depth V2.

**Comment by nagadomi** [comment]
For the moment I am not willing to support diffusion-based depth estimation model in iw3.

**Comment by gituser123456789000** [comment]
Just seeing this now.. came out when I stopped doing 2d to 3d for a while...
As luck or fate would have it, it seems a version 1.1 released just yesterday..

**Reply by gituser123456789000** [reply]
I got it running in ComfyUI.. easy to set that up if you're familiar with ComfyUI already..

I tried directly from the github prior to that and it was a pain, a mess.. needs anaconda, linux.. I think I was very near getting it done, but had no clue what I was missing, so luckily I found the ComfyUI nodes

https://github.com/kijai/ComfyUI-Lotus

from there, you can go into examples and get this .json file
https://github.com/kijai/ComfyUI-Lotus/blob/main/examples/lotus_depth_g_example_01.json
"Download raw file" in the top right area to the left of the pencil icon.
then drag and drop that into ComfyUI
then go to Manager and Install Missing Nodes.. then restart as it says

then you need model(s) and an SD 1.5 or 2.0 VAE
that github has the v1 models. From testing, the v1 models look to pull out background detail better, but look to have white-crushed foreground, so unlikely that the v1 models produce great results. I'll be testing some conversions though using 3DCombine

G v2.1 or whatever other models you want to test are on the main github page:
https://github.com/EnVision-Research/Lotus

these model(s) get added to the ComfyUI/models/diffusion_models folder

then you need a Stable Diffusion 1.5 or 2.0 VAE.. I Googled and use the default one they have in the sample workflow (the .json you drag in)
the VAE goes into ComfyUI/models/vae folder

In use, the v1 models should output white front / black rear..
G v2.1 outputs the opposite though.. black front / white rear.. but luckily they thought of this already and near the end of the workflow is  and "Invert Image" node. Just right-click on that and Bypass. Then v2.1 will output white front / black rear

Review:
It (G v2.1 is my pick so far) seems good with some or many single images, but it's not perfect. It depends on the input.. some images have very obvious incorrect depth.. some have haze/cloudiness. There definitely seems to be a limit to the resolution (just as DepthAnything v2 is 518 for example).. most tests I did were something like 1080p-1440p images, which first impressions (again, depending on the image), turned out good looking.. detailed, consistent, 'correct' looking depth (although not converted yet, so final results uncertain). I tried a 3000x4000 + image though, and that results was a mess. Plenty of incorrect depth, wavy I'd call it, where depth drops in and out. No good at too high resolution, like others. So ideally, we'd have to know the training resolution. Default on their demo is 768, so maybe that's the best resolution of images to use.. not sure.

Single images, hit or miss at best. May or may not be great if you get the right image.. but not good for iw3 I'd say. Too inconsistent. DepthAnything v2 for example, is basically completely consistent.. any image or video, you get an expected result. This Lotus Depth is not that. You have to be more selective with it. Some images may or may not work.. other images definitely won't work.

**Reply by gituser123456789000** [reply]
I tested a little with Lotus D 2.0 also... I wouldn't say any of them are a replacement for DepthAnything for consistent accuracy.

But I did find a use case they seem to be suited for... 

Stable Diffusion Image to Image using the Lotus depth maps via Controlnet

Lotus seems to exaggerate details.. or you could say pulls significantly more depth out of a image compared to other models. Whether this is very accurate or not is a different question, but since it often very clearly pulls out eyes, nose, mouth, clothing, hair, neck, jaw, chin area separation..

**I'm recommending Lotus Depth G v2.1 as very likely the best Stable Diffusion Image to Image model**
